### PR TITLE
OTCP tutorial

### PR DESCRIPTION
**Pull request by EugeneNdiaye**
base draft tutorial otcp

### CODE REVIEW

File: docs/tutorials/misc/300_otcp.ipynb [hunk 1]
```diff
@@ -0,0 +1,780 @@
+{
```
Comment (michalk8) [inline docs/tutorials/misc/300_otcp.ipynb:1]: <p>Same, I'd remove this.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/613/discussion/#comment-2023228930'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (michalk8) [inline docs/tutorials/misc/300_otcp.ipynb:1]: <p>This cell should belong in a new section that is related to model/OTCP fitting and calibration (as opposed to the previous section - data preparation). Please add a new header above.</p><p>Also, we should explain some of the most important arguments for <code>OTCP</code> , especially during <code>fit_transport</code>:<code>epsilon </code> and <code>n_target</code></p><p>Lastly, we should describe what happens in <code>fit_transport</code> a bit as well (solving OT problem, etc.). </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/613/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (michalk8) [inline docs/tutorials/misc/300_otcp.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    skl_model.fit(X_train, Y_train)
</pre><p>The model fitting should be done in a separate cell.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/613/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (michalk8) [inline docs/tutorials/misc/300_otcp.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.    def model(x):
</pre><p>Please add type hints and explain in a docstring why this is needed.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/613/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (michalk8) [inline docs/tutorials/misc/300_otcp.ipynb:1]: <p>I would introduce a new (sub)section above.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/613/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (michalk8) [inline docs/tutorials/misc/300_otcp.ipynb:1]: <p>This needs more explanation (what happens in <code>otcp.predict</code>?  ); would also add references to the paper.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/613/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (michalk8) [inline docs/tutorials/misc/300_otcp.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #7.    ax.scatter(
</pre><p>There's a lot of code repeat for plotting (here, above when just visualizing the data and also below for the forward approach).</p><p>Think it would be great to define a common plotting function which can optionally accept the conformal candidates.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/613/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (michalk8) [inline docs/tutorials/misc/300_otcp.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #34.        x=confidence_regions[idx, :, 1].flatten(),
</pre><p>Don't think <code>.flatten()</code> is needed. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/613/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (michalk8) [inline docs/tutorials/misc/300_otcp.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #42.        label="Contour of Confidence Region",
</pre><p>This can also be removed, it's not being shown.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/613/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (michalk8) [inline docs/tutorials/misc/300_otcp.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #46.    ax.set_xlim(40.68, 40.8)
</pre><p>Same here, would remove the limits.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/613/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (michalk8) [inline docs/tutorials/misc/300_otcp.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    y_candidates = jnp.array(Y_train[:500])
</pre><p>Maybe it would be more interesting here to provide a 2D grid instead of sampling the candidates from <code>y_train</code>.  <span class="ql-cursor">﻿</span></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/613/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (michalk8) [inline docs/tutorials/misc/300_otcp.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #48.    plt.show()
</pre><p>No need for <code>plt.show()</code>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/613/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>
